lib/utils.coffee

```diff
@@ -19,6 +19,8 @@ semver = require('semver')
 fs = Promise.promisifyAll(require('fs'))
 resin = require('resin-sdk-preconfigured')
 
+RESINOS_VERSION_REGEX = /v?\d+\.\d+\.\d+(\.rev\d+)?((\-|\+).+)?/
```

Comment:
This is regex is only a temporary fix for now, so not worth it yet.

This was just discussed on the arch call - this format is now dead: we're pulling & rereleasing ResinOS 2 semver-compatibly, using build metadata (`2.0.0+rev1`) instead. You can blindly parse that as normal semver and get sensible results, and I'm set something up to commonize lots of this logic (either in the SDK or a new module, I'm not sure), and let you easily do sorting etc that takes the rev into account.